Merge Feature/header into setup branch

Adamik10 opened the pull request:
This feature branch updates the header component to comply with the newest figma mockups. The component already existed beforehand - thus changes are minimal.

Ticket: [DDFSOEG-96](https://reload.atlassian.net/browse/DDFSOEG-96)

cableman reviewed (changes requested):
Why is the package-lock file update when the actual change is scss only?

sinejespersen commented on src/stories/header/header.scss:
```diff
@@ -149,6 +149,12 @@
   ::placeholder {
     color: $c-text-secondary-gray;
   }
+
```
I am not sure how this looks "in the wild", but be aware that if it removes the possibility to see if an element is in focus, then it does not comply with WCAG2.1 https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html

Adamik10 replied:
yes @sinejespersen. In the mockups, focus state is shown by changing the background color of the input field to white without it being outlined. So the whole field changes color from:
![image](https://user-images.githubusercontent.com/28546954/169246371-af9fee4d-fea4-4c09-ba41-4bdf40f6e38a.png)
to:
![image](https://user-images.githubusercontent.com/28546954/169246192-822e6b95-4f76-4956-97b5-282669d3690a.png)

sinejespersen replied:
Sure is, thanks.

Adamik10 commented:
> Why is the package-lock file update when the actual change is scss only?

@cableman It is because I didn't commit these things with previous changes, where we introduced clsx package and removed engines. And THAT's because I thought package-lock.json should have been a part of git ignore, which now I know it's ofc indeed not. I'm also fairly new to git, so it is entirely possible that these changes already lay somewhere on another branch, and I should have just rebased them into here.
